src/Engine.cpp

```diff
@@ -607,11 +607,62 @@ static void opt_version() {
 		"...please don't sue us." << std::endl;
 }
 
-bool Engine::parseCommandLine(int argc, char *argv[]) {
+bool Engine::parseCommandLine(int argc, char* argv[])
+{
+
 	// variables for command-line flags
 	int optret;
 	std::vector<std::string> data_vec;
 
+
+
+#ifdef _WIN32
+
+	char path[MAX_PATH];
+
+	HRESULT hr = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED |
+		COINIT_DISABLE_OLE1DDE);
+	IFileOpenDialog* fo;
+
+	hr = CoCreateInstance(CLSID_FileOpenDialog, NULL, CLSCTX_ALL,
+		IID_IFileOpenDialog, reinterpret_cast<void**>(&fo));
+
+	fo->SetOptions(FOS_PICKFOLDERS);
+
+	if (SUCCEEDED(hr))
+	{
+		// Show the Open dialog box.
+		hr = fo->Show(NULL);
+
+		// Get the file name from the dialog box.
+		if (SUCCEEDED(hr))
+		{
+			IShellItem* pItem;
+			hr = fo->GetResult(&pItem);
+			if (SUCCEEDED(hr))
+			{
+				PWSTR pszFilePath;
+				hr = pItem->GetDisplayName(SIGDN_FILESYSPATH, &pszFilePath);
+				wcstombs(path, pszFilePath, MAX_PATH);
```

Comment:
Can you use codecvt instead? wctombs has issues with locales (might not return UTF-8) and passing around raw pointers make me nervous (for instance, the max path size in characters can be bigger than the max path size in wide-characters due to UTF-8 multi-byte codepoints; and wctombs isn't guaranteed to put on a null byte at the end).

There's an example usage of codecvt in userlocale.cpp that does exactly what we want here: safely convert a UTF-16 pointer into a UTF-8 std::string